UI/js/controllers.js

```diff
@@ -126,35 +126,40 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
   //use the Terms factory as a promise. Add returned data to the scope
 
   Terms.getTerms(termsUrl).then(function (data) {
-    // the ESB might return a sinble object rather than an array, turn it into an array
-    if (data.Result.length === undefined ){
-      data.Result = [].concat(data.Result);
+    if (data.failure) {
+      $scope.$parent.term  = $rootScope.lang.termFailure
     }
-    if (data.Result.length !==0)  {
+    else {
+      // the ESB might return a sinble object rather than an array, turn it into an array
+      if (data.Result.length === undefined ){
+        data.Result = [].concat(data.Result);
+      }
+      if (data.Result.length !==0)  {
 
-      $scope.terms = data.Result;
-      $scope.$parent.term = data.Result[0].TermDescr;
-      $scope.$parent.termId = data.Result[0].TermCode;
+        $scope.terms = data.Result;
+        $scope.$parent.term = data.Result[0].TermDescr;
+        $scope.$parent.termId = data.Result[0].TermCode;
 
-      $scope.courses = [];
-      $scope.loading = true;
-      var url = 'courses/' + $rootScope.user + '.json?TERMID=' + $scope.$parent.termId;
+        $scope.courses = [];
+        $scope.loading = true;
+        var url = 'courses/' + $rootScope.user + '.json?TERMID=' + $scope.$parent.termId;
 
-    //use the Courses factory as a promise. Add returned data to the scope.
+      //use the Courses factory as a promise. Add returned data to the scope.
 
-      Courses.getCourses(url).then(function (data) {
-        if (data.failure) {
-          $scope.courses.errors = data;
-          $scope.loading = false;
-        } else {
-          $scope.courses = data;
-          $scope.loading = false;
-        }
-        $('.colHeader small').append($('<span id="done" class="sr-only">' + $scope.courses.length + ' courses </span>'));
-      });
-    } else {
-      $scope.$parent.term  = 'You do not seem to have courses in any terms we know of.'
-    }  
+        Courses.getCourses(url).then(function (data) {
+          if (data.failure) {
+            $scope.courses.errors = data;
+            $scope.loading = false;
+          } else {
+            $scope.courses = data;
+            $scope.loading = false;
+          }
+          $('.colHeader small').append($('<span id="done" class="sr-only">' + $scope.courses.length + ' courses </span>'));
```

Comment:
should we append the term information into this line also to make the message more specific?